sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -87,12 +89,19 @@ RT::PiProgram ProgramManager::getBuiltOpenCLProgram(OSModuleHandle M,
   std::shared_ptr<context_impl> Ctx = getSyclObjImpl(Context);
   std::map<OSModuleHandle, RT::PiProgram> &CachedPrograms =
       Ctx->getCachedPrograms();
-  RT::PiProgram &Program = CachedPrograms[M];
-  if (!Program) {
-    DeviceImage *Img = nullptr;
-    Program = loadProgram(M, Context, &Img);
-    build(Program, Img->BuildOptions);
-  }
+  auto It = CachedPrograms.find(M);
+  if (It != CachedPrograms.end())
+    return It->second;
+
+  DeviceImage *Img = nullptr;
+  using PiProgramT = std::remove_pointer<RT::PiProgram>::type;
```

Comment:
```suggestion
  using PiProgramT = remove_pointer_t<RT::PiProgram>;
```
from .../detail/type_traits.hpp

Reply:
Done.

Reply:
Please include the file that contains remove_pointer_t explicitly.